### ISSUE TITLE
ExecuteQuery() throws exception

### ISSUE THREAD

**Issue by furieuxjk** [PowerShell/PowerShell#11871]
I don't know if this is the correct place to report this issue, but script works with PowerShell 5.1, but not with PowerShell 6 and 7.

# Steps to reproduce

```powershell
Add-Type -Path .\Microsoft.SharePoint.Client.dll
Add-Type -Path .\Microsoft.SharePoint.Client.Runtime.dll
$siteUrl = "https://sharepoint.example.com/sites/testsite"
$username = "DOMAIN\user"
$password = "SomePassword"
$ctx = New-Object Microsoft.SharePoint.Client.ClientContext($siteUrl)
$ctx.Credentials = New-Object System.Net.NetworkCredential($username, (ConvertTo-SecureString $password -AsPlainText -Force))
$list = $ctx.Web.Lists.GetByTitle('ListName')
$ctx.Load($list)
$ctx.ExecuteQuery()
$list.ItemCount
```

# Expected behavior

```none
3
```

# Actual behavior

```none
Exception calling "ExecuteQuery" with "0" argument(s): "The remote server returned an error: (400) Bad Request."
At line:1 char:1
+ $ctx.ExecuteQuery()
+ ~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
+ FullyQualifiedErrorId : WebException
```

# Environment data

```none
Name                           Value
----                           -----
PSVersion                      6.2.4
PSEdition                      Core
GitCommitId                    6.2.4
OS                             Microsoft Windows 10.0.18363
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

and

Name                           Value
----                           -----
PSVersion                      7.0.0-rc.2
PSEdition                      Core
GitCommitId                    7.0.0-rc.2
OS                             Microsoft Windows 10.0.18363
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by iSazonov** [comment]
@furieuxjk Sharepoint module is not in the repo. Please report on UserVoice site or use other channels to communicate with Sharepoint team.

**Comment by furieuxjk** [comment]
Ok, thanks!

**Comment by furieuxjk** [comment]
So, according to UserVoice:
https://sharepoint.uservoice.com/forums/329220-sharepoint-dev-platform/suggestions/16585795-support-net-core-with-csom

looks like now SharePoint module should work. But it doesn't. Or I am doing something wrong.

I used Visual Studio's NuGet package Manager Console to get CSOM:
`Install-Package Microsoft.SharePointOnline.CSOM -Version 16.1.20211.12000`

In the end in my Visual Studio project I got this subfolder:
`\packages\Microsoft.SharePointOnline.CSOM.16.1.20211.12000\lib`

which has these subfolders:
javascript
net40-full
net45
netcore45
netstandard2.0
portable-net45+netcore45+wpa81
wpa

I tried to use these files:
Microsoft.SharePoint.Client.Portable.dll
Microsoft.SharePoint.Client.Runtime.Portable.dll

from "net45" folder and "portable-net45+netcore45+wpa81", but non of them work - I got the same error as in the first post.
Basically I have tried those two dll files from all folders and my script works from Windows PowerShell 5.1, but doesn't work in PowerShell 7.

What am I doing wrong?

**Comment by vexx32** [comment]
Are there usable assemblies in `netstandard2.0`?

**Comment by furieuxjk** [comment]
Nope, if I use assemblies from netstandard2.0, then even in Windows PowerShell 5.1 it doesn't work and I got strange permission error:

> Exception calling "ExecuteQuery" with "0" argument(s): "The remote server returned an error: (403) Forbidden."
At C:\Temp2\sp_list_test2.ps1:25 char:1
$ctx.executeQuery()

> CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
FullyQualifiedErrorId : WebException

>The collection has not been initialized. It has not been requested or the request has not been executed. It may need to be explicitly requested.

**Comment by iSazonov** [comment]
I think it is better to ask on UserVoice site or MSFT Sharepoint forum.

**Comment by furtivesock** [comment]
(sorry for bumping)
Hello @furieuxjk, were you able to find any solution for this issue?

**Comment by furieuxjk** [comment]
> (sorry for bumping)
> Hello @furieuxjk, were you able to find any solution for this issue?

Hello.

No, I left this for later time and still didn’t come back to this issue.

You have the same problem?

**Comment by furtivesock** [comment]
> > (sorry for bumping)
> > Hello @furieuxjk, were you able to find any solution for this issue?
> 
> Hello.
> 
> No, I left this for later time and still didn’t come back to this issue.
> 
> You have the same problem?

Yes, I have the same problem.. Never mind.. I use both PowerShell 7 and PowerShell 5.1 when it's needed.

**Comment by azpik** [comment]
I have the same issue. Any thoughts how this can be resolved?
